Downsample/Downsample.cpp

```diff
@@ -0,0 +1,196 @@
+/*
+Import a ZDF point cloud and downsample it.
+*/
+
+#include <Zivid/CloudVisualizer.h>
+#include <Zivid/Zivid.h>
+#include <Eigen/Dense>
+#include <Eigen/Core>
+
+#include <iostream>
+Eigen::MatrixXf sumline(Eigen::MatrixXf&, int, int);
+Eigen::MatrixXf gridsum(Eigen::MatrixXf&, int);
+float myround(float); 
+float isnanmask(float);
+float nantozero(float);
+Zivid::PointCloud downsample(Zivid::PointCloud&, int);
+
+int main()
+{
+	try
+	{
+		Zivid::Application zivid;
+
+		std::string Filename = "Zivid3D.zdf";
+		std::cout << "Reading " << Filename << " point cloud" << std::endl;
+		Zivid::Frame frame = Zivid::Frame(Filename);
+
+		auto pointCloud = frame.getPointCloud();
+		auto downSamplingFactor = 4;
+
+		Zivid::PointCloud pointCloudDownsampled(pointCloud.height()/downSamplingFactor, pointCloud.width()/downSamplingFactor);
+
+		pointCloudDownsampled = downsample(pointCloud, downSamplingFactor);
+
+		std::cout << "Setting up visualization" << std::endl;
+		{
+			Zivid::CloudVisualizer vis;
+			zivid.setDefaultComputeDevice(vis.computeDevice());
+
+			std::cout << "Displaying the original point cloud" << std::endl;
+			vis.showMaximized();
+			vis.show(pointCloud);
+			vis.resetToFit();
+
+			std::cout << "Running the visualizer. Blocking until the window closes" << std::endl;
+			vis.run();
+		}
+		{
+			Zivid::CloudVisualizer vis;
+			zivid.setDefaultComputeDevice(vis.computeDevice());
+
+			std::cout << "Displaying the downsampled point cloud" << std::endl;
+			vis.showMaximized();
+			vis.show(pointCloudDownsampled);
+			vis.resetToFit();
+
+			std::cout << "Running the visualizer. Blocking until the window closes" << std::endl;
+			vis.run();
+		}
+	}
+
+	catch (const std::exception &e)
+	{
+		std::cerr << "Error: " << Zivid::toString(e) << std::endl;
+		return EXIT_FAILURE;
+	}
+}
+
+Eigen::MatrixXf sumline(Eigen::MatrixXf& matrix, int downSamplingFactor)
+{
+	Eigen::MatrixXf reshapedMatrix = matrix.reshaped(downSamplingFactor, (int)matrix.rows() * (int)matrix.cols() / downSamplingFactor);
+	
+	std::function<float(float)> nan_to_zero_wrap = nantozero;
+
+	reshapedMatrix = reshapedMatrix.unaryExpr(nan_to_zero_wrap);
+
+	return reshapedMatrix.colwise().sum().reshaped((int)matrix.rows() / downSamplingFactor, (int)matrix.cols()).transpose();
+}
+
+Eigen::MatrixXf gridsum(Eigen::MatrixXf& r, int downSamplingFactor)
+{
+	return sumline(sumline(r, downSamplingFactor), downSamplingFactor);
+}
+
+float myround(float x)
+{
+	return std::round(x);
+}
+
+float isnanmask(float x)
```

Comment:
What's this function for? Would it make the code simpler if you remove it and replace all usages with `!std::isnan` directly instead?

If you do keep this function, the function body can be simplified to `return !std::isnan(x)`.

I also find it surpising that a `is...` function returns `float`, as `is...` sounds like predicate that should return `bool`.